app/controllers/post_voting/votes_controller.rb

```diff
@@ -133,37 +133,45 @@ def ensure_can_see_comment!(comment)
     end
 
     def ensure_can_vote(votable)
-      error_message = nil
-      error_message_params = {}
+      raiseError("post.post_voting.errors.vote_archived_topic") if votable.topic.archived?
+
+      raiseError("post.post_voting.errors.vote_closed_topic") if votable.topic.closed?
 
       if votable.user_id == current_user.id
-        error_message = "post.post_voting.errors.self_voting_not_permitted"
-      elsif votable.class.name == "Post"
+        raiseError("post.post_voting.errors.self_voting_not_permitted")
+      end
+
+      if votable.class.name == "Post"
         direction = vote_params[:direction] || QuestionAnswerVote.directions[:up]
         if QuestionAnswerVote.exists?(
              votable: votable,
              user_id: current_user.id,
              direction: direction,
            )
-          error_message = "vote.error.one_vote_per_post"
+          raiseError("vote.error.one_vote_per_post")
         elsif !PostVoting::VoteManager.can_undo(votable, current_user)
-          error_message = "vote.error.undo_vote_action_window"
-          error_message_params = { count: SiteSetting.post_voting_undo_vote_action_window.to_i }
+          raiseError(
+            "vote.error.undo_vote_action_window",
+            { count: SiteSetting.post_voting_undo_vote_action_window.to_i },
+          )
         end
-      elsif votable.class.name == "QuestionAnswerComment"
-        if QuestionAnswerVote.exists?(votable: votable, user: current_user)
-          error_message = "vote.error.one_vote_per_comment"
+
+        if votable.class.name == "QuestionAnswerComment" &&
+             QuestionAnswerVote.exists?(votable: votable, user: current_user)
+          raiseError("vote.error.one_vote_per_comment")
         end
       end
+    end
 
-      if error_message.present?
-        raise Discourse::InvalidAccess.new(
-                nil,
-                nil,
-                custom_message: error_message,
-                custom_message_params: error_message_params,
-              )
-      end
+    private
+
+    def raiseError(error_message, error_message_params = nil)
```

Comment:
What in the js naming!? 🤪 

```suggestion
    def raise_error(error_message, error_message_params = nil)
```

Reply:
:joffrey_facepalm: